src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,39 @@
 package mate.academy;
 
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final ConcurrentLinkedQueue<EventListener> listeners;
+    private final ExecutorService executorService;
+    private final int cores;
 
+    {
+        listeners = new ConcurrentLinkedQueue<>();
+        cores = Runtime.getRuntime().availableProcessors();
     }
 
-    public void deregisterListener(EventListener listener) {
+    public EventManager() {
+        executorService = Executors.newFixedThreadPool(cores > 1 ? cores / 2 : 1);
```

Comment:
why did you choose to have the number of threads twice less than the number of cores?